lib/BIGSdb/Plugins/PhyloViz.pm

```diff
@@ -0,0 +1,415 @@
+#PhyloViz.pm - phylogenetic inference and data visualization for sequence based typing methods for BIGSdb
+#Written by Emmanuel Quevillon
+#Copyright (c) 2016, Institut Pasteur, Paris
+#E-mail: tuco@pasteur.fr
+#
+#This file is part of Bacterial Isolate Genome Sequence Database (BIGSdb).
+#
+#BIGSdb is free software: you can redistribute it and/or modify
+#it under the terms of the GNU General Public License as published by
+#the Free Software Foundation, either version 3 of the License, or
+#(at your option) any later version.
+#
+#BIGSdb is distributed in the hope that it will be useful,
+#but WITHOUT ANY WARRANTY; without even the implied warranty of
+#MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#GNU General Public License for more details.
+#
+#You should have received a copy of the GNU General Public License
+#along with BIGSdb.  If not, see <http://www.gnu.org/licenses/>.
+package BIGSdb::Plugins::PhyloViz;
+use strict;
+use warnings;
+use 5.010;
+use parent qw(BIGSdb::Plugin);
+use List::MoreUtils qw(uniq);
+use Log::Log4perl qw(get_logger);
+my $logger = get_logger('BIGSdb.Plugins');
+use Apache2::Connection ();
+use constant DEBUG => 0;
+
+sub get_attributes {
+	my ($self) = @_;
+	my %att = (
+		name        => 'PhyloViz',
+		author      => 'Emmanuel Quevillon',
+		affiliation => 'Institut Pasteur, Paris',
+		email       => 'tuco@pasteur.fr',
+		description => 'Creates phylogenetic inference and data visualization for sequence based typing methods',
+	        category    => 'Analysis',
+		buttontext  => 'PhyloViz',
+	        # Set menutext to empty string to not display it on index page (See BIGSdb::IndexPage line 329)
+	        menutext    => '',
+		module      => 'PhyloViz',
+		version     => '0.0.1',
+		dbtype      => 'isolates',
+		section     => 'analysis,postquery',
+		url         => "$self->{'config'}->{'doclink'}/data_analysis.html#phyloviz",
+		input       => 'query',
+		system_flag => 'PhyloViz',
+		requires    => 'ref_db,js_tree',
+		help        => 'tooltips',
+		order       => 33,
+	);
+	return \%att;
+}
+
+sub run {
+    my ($self) = @_;
+
+    my $q = $self->{'cgi'};
+    say q(<h1>PhyloViz: phylogenetic tree vizualisation</h1>);
+
+    # Get the list of isolates from query
+    my $query_file = $q->param('query_file');
+    my $qry_ref    = $self->get_query($query_file);
+
+    if( ref $qry_ref ne 'SCALAR' ){
+	say q(<div class="box" id="statusbad"><p>Can not retrieve id list. Please repeat your query</p></div>);	
+	return;
+    }
+    $self->rewrite_query_ref_order_by($qry_ref);
```

Comment:
Does the order of isolates matter for the analysis? - if not, I think you can remove this.